pyrit/prompt_target/openai/openai_sora_target.py

```diff
@@ -83,71 +110,73 @@ def _should_retry_video_download(response: httpx.Response) -> bool:
+                Defaults to "1280x720" which works with both API versions.
+                Sora-1 supported resolutions are "360x360", "640x360", "480x480", "854x480",
+                "720x720", "1280x720", "1080x1080", "1920x1080".
+                Sora-2 supported resolutions are "720x1280", "1280x720", "1080x1920", "1920x1080".
+            n_seconds (int, Optional): The duration of the generated video (in seconds).
+                Defaults to 4 (compatible with both APIs).
+                Sora-1 supports up to 20 seconds (10 seconds max for 1080p resolutions).
+                Sora-2 supports exactly 4, 8, or 12 seconds.
+            n_variants (int, Optional): Number of video variants to generate. Defaults to 1.
+                Only supported by Sora-1 API (ignored for Sora-2).
             output_filename (str, Optional): The name of the output file for the generated video.
-                Note: DO NOT SET if using target with PromptSendingAttack. The default filename
-                is {job_id}_{generation_id}.mp4 as returned by the model.
-
-        Raises:
-            ValueError: If video constraints are not met for the specified resolution.
+                Note: DO NOT SET if using target with PromptSendingAttack.
         """
-        super().__init__(**kwargs)
-
-        dimensions = resolution_dimensions or self.DEFAULT_RESOLUTION_DIMENSIONS
-        temp = dimensions.split("x")
-        self._height = temp[1]
-        self._width = temp[0]
-
-        self._n_seconds = n_seconds or self.DEFAULT_N_SECONDS
-        self._n_variants = n_variants or self.DEFAULT_N_VARIANTS
-        self._api_version = api_version or self.DEFAULT_API_VERSION
-
-        # Validate input based on resolution dimensions
-        self._validate_video_constraints(
-            resolution_dimensions=dimensions,
-            n_seconds=self._n_seconds,
-            n_variants=self._n_variants,
-        )
+        # Parse resolution
+        if "x" not in resolution_dimensions:
+            raise ValueError(
+                f"Invalid resolution format: '{resolution_dimensions}'. "
+                "Expected format: 'WIDTHxHEIGHT' (e.g., '1280x720')"
+            )
+        dimensions = resolution_dimensions.split("x")
+        if len(dimensions) != 2:
+            raise ValueError(
+                f"Invalid resolution format: '{resolution_dimensions}'. "
+                "Expected format: 'WIDTHxHEIGHT' (e.g., '1280x720')"
+            )
+        self._height = dimensions[1]
+        self._width = dimensions[0]
 
+        self._n_seconds = n_seconds
+        self._n_variants = n_variants
         self._output_filename = output_filename
+        self._params: Dict[str, Any] = {}  # Initialize params dict
 
-        self._params = {}
-        self._params["api-version"] = self._api_version
+        # Initialize parent class first to get endpoint
+        super().__init__(**kwargs)
 
```

Comment:
```suggestion
@property
def version(self) -> int:
    return 1 if "v1" in self._endpoint else 2
```
May be more ergonomic and extensible than _detected_api_version (e.g. if self.version > 1 ...) but also more brittle; maybe setting the method below to a @ property would get the best of both worlds?

Reply:
Resolving as self._api_version seems fine for internal class use